### PR TITLE
remove redundant package data inclusion

### LINKED ISSUE #644: Revisit how we package and use non-python files
Thinking about this because I needed to add a translation message catalog file to the `natcap.invest` package.
https://setuptools.readthedocs.io/en/latest/userguide/datafiles.html

We currently use both `MANIFEST.in` and `package_data`. [This article](https://www.remarkablyrestrained.com/python-setuptools-manifest-in/) argues for replacing `MANIFEST.in` with a list of checked-in files that you can get from `setuptools_scm`, which we already use for something else. [The setuptools docs](https://setuptools.readthedocs.io/en/latest/userguide/extension.html#adding-support-for-revision-control-systems) detail that option.

- Do we need to use both `MANIFEST.in` and `package_data`?
- Would it simplify things to use only one of them?
- Should we be accessing package data using the setuptools `ResourceManager` API like they [recommend](https://setuptools.readthedocs.io/en/latest/userguide/datafiles.html#accessing-data-files-at-runtime), instead of `__file__`? This is relevant for the existing `reporting_data` and also for translation catalogs.

### PR DESCRIPTION
**Pull request by emlys**
## Description
Fixes #644

Because `setuptools_scm` is installed in the build environment (pyproject.toml), all tracked files are automatically included in the sdist.
Because `setuptools_scm` is installed in the build environment (pyproject.toml) AND `include_package_data=True`, all tracked files in `src/` are automatically included in the wheel.
So, the `package_data` and `MANIFEST.in` `include` statements were redundant.
`MANIFEST.in` is still needed to exclude top-level directories from the sdist. `exclude_package_data` only works within the package.

## Checklist
- [ ] ~Updated HISTORY.rst (if these changes are user-facing)~
- [ ] ~Updated the user's guide (if needed)~
- [ ] ~Tested the affected models' UIs (if relevant)~

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -157,10 +157,6 @@ jobs:
 
       - name: Install from source distribution
         run : |
-          # Prevent pip from thinking that CWD is a natcap.invest
-          # installation. It's not.
-          rm -r natcap.invest.egg-info
```
Comment (emlys) [inline .github/workflows/build-and-test.yml:162]: This doesn't seem to be a problem any more, maybe because `natcap.invest.egg-info` is now in `src/`.

Comment (phargogh) [reply .github/workflows/build-and-test.yml:162]: Neat!  I wonder when they made that change.

**Review by phargogh — CHANGES_REQUESTED** [review]
Thanks @emlys !  It's really nice to finally reconcile these and update to the latest conventions.

I'm only requesting changes because I wonder what you think about also `prune`ing `./workbench/` from the source distribution.  Let me know what you think!

Oh, one other thing that comes to mind is that since we're changing what's in the source and wheel distributions, do you think it's worth mentioning that in `HISTORY`?

File: MANIFEST.in [hunk 2]
```diff
@@ -1,13 +1,5 @@
...
-include LICENSE.txt
-include pyproject.toml
-include requirements.txt
-include requirements-dev.txt
-include requirements-gui.txt
 prune .github
 prune ci
+prune doc/users-guide
 prune exe
 prune installer
```
[the remark this answers is not in the document]
Comment (phargogh) [reply MANIFEST.in]: Ha, whoops!  I'm a little surprised that `bitbucket-pipelines.yml` was still in there.   Thanks for removing that (and the others too) and excluding these other files as well.

**Pull request merged**